### PR TITLE
Domain Randomization API

### PR DESCRIPTION
**Pull request by bhairavmehta95**
Switching to config files for domain randomization.

Some of the graphics stuff will need some reorganizing, but this is a pretty basic addition - looking for feedback. If we like this, we can merge and I can do the texture-based randomization the same way.

@liampaull

### CODE REVIEW

File: gym_duckietown/randomization/config/default.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "horz_mode": {
```
Comment (liampaull) [inline gym_duckietown/randomization/config/default.json]: is it documented somewhere what these numbers mean?

Comment (liampaull) [reply gym_duckietown/randomization/config/default.json]: i.e. how they are applied in each case? or should one go and look at the code to see how they are applied?

**Comment by liampaull** [comment]
@bhairavmehta95 have a look at the results of the circle ci test? is something missing? Otherwise I'm basically fine to merge but we should write down somewhere, either in here or in the docs, what are the variables that you can randomize and what the numbers mean. The API looks good though

**Comment by bhairavmehta95** [comment]
Yeah so the CircleCI tests fail because of a variable reference:

https://github.com/duckietown/gym-duckietown/blob/master/learning/reinforcement/pytorch/train_reinforcement.py#L64-L84

It says that the variable is unreferenced, but it gets set before it actually would ever step into that if statement. (aka it's fine)

And yes - this PR was more about getting comments on the API - I can write up a little README to put inside the `randomization/` folder.

Unfortunately though, while we can _qualitatively_ describe what each variable does, certain things are just uninterpretable. i.e the global light position will always be a 3D -point in space, and there's not an intuitive range that says "this range of 3D coordinates makes sense while this other range doesn't."

I'll write something up though, atleast as a start.

File: gym_duckietown/randomization/config/default.json (hunk 1 above)
Comment (bhairavmehta95) [reply gym_duckietown/randomization/config/default.json]: I will have a README with a qualitative description, as well as a reference to the line number where each item is applied. I think that should handle it.

File: README.md [hunk 2]
```diff
@@ -65,6 +66,11 @@ control your robot remotely with the `Duckiebot-v0` environment, you will need t
 install the software found in the [duck-remote-iface](https://github.com/maximecb/duck-remote-iface)
```
[the remark this answers is not in the document]
Comment (bhairavmehta95) [reply README.md]: I'm not sure what we'd change it to at the moment, so leaving.

**Comment by liampaull** [comment]
ok sounds great

**Pull request merged**